content/guides/10-minute-tutorial.md

```diff
@@ -792,12 +793,281 @@ Feature: Is it Friday yet?
+  @today = 'Friday'
+end
+
+# Then is a shared function
+When("I ask whether it's Friday yet") do
+  @actual_answer = is_it_friday?(@today)
+end
+
+# Then is a shared function
+Then("I should be told {string}") do |expected_answer|
+  expect(@actual_answer).to eq(expected_answer)
+end
+```
+{{% /block %}}
+
+Run Cucumber again:
+
+{{% block "javascript" %}}
+```shell
+......
+2 scenarios (2 passed)
+6 steps (6 passed)
+0m00.002s
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```shell
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario: Sunday isn't Friday        # features/is_it_friday_yet.feature:4
+    Given today is Sunday              # features/step_definitions/stepdefs.rb:8
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:17
+    Then I should be told "Nope"       # features/step_definitions/stepdefs.rb:22
+
+  Scenario: Friday is Friday           # features/is_it_friday_yet.feature:9
+    Given today is Friday              # features/step_definitions/stepdefs.rb:12
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:17
+    Then I should be told "TGIF"       # features/step_definitions/stepdefs.rb:22
+
+2 scenarios (2 passed)
+6 steps (6 passed)
+0m0.040s
+```
+{{% /block %}}
+
+# Using variables and examples
+So, we all know that there are more days in the week than just Sunday and Friday. Let's update our scenario to use variables and evaluate more possibilities. We'll use variables and examples to evaluate Friday, Sunday, and anything else!
+
+Update the `is-it-friday-yet.feature` file. Notice how we go from `Scenario` to `Scenario Outline` when we start using `Examples`.
+```gherkin
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario Outline: Today is or is not Friday
+    Given today is <day>
+    When I ask whether it's Friday yet
+    Then I should be told <answer>
+
+  Examples:
+    | day | answer |
+    | "Friday" | "TGIF" |
+    | "Sunday" | "Nope" |
+    | "anything else!" | "Nope" |
+```
+
+{{% block "javascript" %}}
+Update the `stepdefs.js` file to use the `<day>` and `<answer>` variables.
+```javascript
+const assert = require('assert');
+const { Given, When, Then } = require('cucumber');
+
+function isItFriday(today) {
+  return (today === "Friday") ? "TGIF" : "Nope";
+}
+
+Given('today is {string}', function (givenDay) {
+  this.today = givenDay;
+});
+
+When('I ask whether it\'s Friday yet', function () {
+  this.actualAnswer = isItFriday(this.today);
+});
+
+Then('I should be told {string}', function (expectedAnswer) {
+  assert.equal(this.actualAnswer, expectedAnswer);
+});
+
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+Update the `stepdefs.rb` file to use the `<day>` and `<answer>` variables.
+```ruby
+module FridayStepHelper
+  def is_it_friday?(day)
+    answer = (day == 'Friday') ? 'TGIF' : 'Nope'
+  end
+end
+World FridayStepHelper
+
+Given("today is {string}") do |given_day|
```

Comment:
this and the Assertion can be single quotes.

Reply:
I don't have strong feelings one way or another on this, but the [current 10 minute tutorial](https://docs.cucumber.io/guides/10-minute-tutorial/) is formatted with " for these.

Reply:
There are good reasons for keeping these double.